content/tutorials/guides/how-to-collect-product-variations.md

```diff
@@ -0,0 +1,274 @@
+```
+
+### Use case 1: Collect product variation information - all levels
+
+#### 1. Collect product models
+##### 1.1 You are following the App workflow?
+
+In the guided tutorial How to get families and attributes, we have stored a family_code_list. It’s time to use it!
+
+```php [activate:PHP]
+
+function getProductModels(): array
+{
+    $client = getApiClient();
+
+    $maxProductsPerPage = 100;
+    $maxFamiliesPerQuery = 3;
+    $locales = 'fr_FR';
+    $scope = 'ecommerce';
+
+    $familyCodeChunks = array_chunk(getFamilyCodes(), $maxFamiliesPerQuery);
+
+    $apiUrl = '/api/rest/v1/product-models?'
+        . 'locales=%s'
+        . '&scope=%s'
+        . '&search={"family":[{"operator":"IN","value":%s}]}'
+        . '&limit=' . $maxProductsPerPage;
+
+
+    // Collect product models from paginated API
+    $productModels = [];
+    foreach ($familyCodeChunks as $familyCodes) {
+        $response = $client->get(sprintf($apiUrl, $locales, $scope, json_encode($familyCodes)));
+        $data = json_decode($response->getBody()->getContents(), true);
+        $productModels[] = $data['_embedded']['items'];
+    }
+
+    $productModels = array_merge(...$productModels);
+
+    saveProductModels($productModels);
+
+    return $productModels;
+}
+
+```
+
+##### 1.2 You are not following the App workflow?
+Simply get the attribute type by requesting the API
+
+```php [activate:PHP]
+
+function getProductModels(): array
+{
+    $client = getApiClient();
+
+    $maxProductsPerPage = 100;
+    $locales = 'fr_FR';
+    $scope = 'ecommerce';
+
+    $apiUrl = '/api/rest/v1/product-models?'
+        . 'locales=%s'
+        . '&scope=%s'
+        . '&limit=' . $maxProductsPerPage;
+
+    // Collect product models from paginated API
+    $response = $client->get(sprintf($apiUrl, $locales, $scope));
+    $data = json_decode($response->getBody()->getContents(), true);
+
+    $productModels = [];
+    $productModels[] = $data['_embedded']['items'];
+    while (array_key_exists('next', $data['_links'])) {
+        $response = $client->get($data['_links']['next']['href']);
+        $data = json_decode($response->getBody()->getContents(), true);
+        $productModels[] = $data['_embedded']['items'];
+    }
+
+    $productModels = array_merge(...$productModels);
+
+    saveProductModels($productModels);
+
+    return $productModels;
+}
+
+```
+
+#### 2. Process product model
+##### 2.1. Parse and store the product model
+Parse and store a product or a product model is definitely the same thing. Please have a how to our guided tutorial **“How to get product information”**.
+
+##### 2.2. Parse and store the product model
+###### 2.2.1 You are following the App workflow?
+
+Good news: you already store the family variant in the guided tutorial **“How to get families and attributes”**. Go ahead!
+
+###### 2.2.2 You are not following the App workflow?
+Query the API.
+
+```php [activate:PHP]
+
+function getFamilyVariants(): array
+{
+    $client = getApiClient();
```

Comment:
I guess this is the initialization part? If yes, you should change your step 0 and define `getApiClient()`